guides/v2.2/contributor-guide/contributing_dod.md

```diff
@@ -57,74 +57,115 @@ Any backward-incompatible changes must also be recorded in the accompanying docu
+* For UI-related code, there are some circumstances where it is not possible to cover changes with unit or integration tests. In these cases the appropriate type of coverage would be a functional test but this is a last resort only. 
+  However, aside from those cases UI functional tests should only be used to cover P0/P1 testing scenarios. If there are no scenarios available to make this decision, a product owner must be consulted for approval of new tests.
+  
+  If there is a scenario new or old that isn't classified as a P0/P1, it can be discussed with a product owner to get it elevated if it is important enough to deserve a dedicated test.
 
 See [Functional Tests][2].
 
 ### Unit Tests
 
-Cover classes and methods that:
+There are a small number of use cases for unit tests in Magento. The nature of our code and backwards compatibility practices make it increasingly hard to write and maintain high-quality unit tests.
+Many of them end up being replaced by integration tests or are practically useless from the beginning due to how much mocking is needed to make them pass.
+
+For these reasons, most of the time an integration test is likely the preferred test type but here are some cases when you would still want to use them:  
 
-* Have little to no dependencies
-* Do not interact with resources like database, file system, 3rd party systems etc.
-* Underlying functionality cannot be properly covered indirectly by an integration test
-* Can be covered by a [black box test](https://en.wikipedia.org/wiki/Black-box_testing)
+Classes and methods that:
+
+* Have little to no dependencies. 
+  - The test must not be directly coupled to the collaborators of the test subject. If the collaborators are solely responsible for the behavior of the tested subject, it won't be a good unit test. 
+* Do not interact with resources like database, file system, 3rd party systems etc. 
+* Can be covered by a black box test
+  - The test must not be a mirror of the code. Instead the test must validate the contract of the method or class is upheld by the implementation. Specific implementations may be weak in certain areas so specific corner cases could cover that logic while still being a black box test.  
+* Would not be easily foreseen to have dependencies in the future.
+  - This can be hard to predict but a good example of when it would be unlikely is a simple utility class that performs some standalone operation. By contrast, a helper would be very likely to have dependencies by nature. 
 
 Examples:
 
-* Utils (like `\Magento\Framework\Math\Random`)
-* Simple classes that can also be used (at least initiated) independently (like `\Magento\Framework\Api\SortOrder`)
-* Algorithms that perform calculation, parsing
+* Utility classes (like `\Magento\Framework\Math\Random`).
+* Simple classes that can also be used independently (like `\Magento\Framework\Api\SortOrder`).
+* Simple logic within a Data Transfer Object.
```

Comment:
Are there examples? Most likely, DTOs should not have any logic.

Reply:
To be honest I was having a hard time with giving good examples for unit tests in Magento and any concrete examples would likely reference undesirable design. I can just remove this example if it would be better unless you have more to add?

Reply:
I'm ok with removing this one.